### PR TITLE
Feature/add booking creating

### PR DESCRIPTION
**Pull request by SolovyovDmitry**
Create booking service according to current standarts and add creating booking

### CODE REVIEW

File: src/BookingService.Data/Booking/BookingRepository.cs [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+      IDataProvider provider,
+      ILogger<BookingRepository> logger)
+    {
+      _provider = provider;
+      _logger = logger;
+    }
+
+    public async Task CreateAsync(DbBooking booking)
+    {
+      if (booking == null)
```
Comment (VLiamin) [inline src/BookingService.Data/Booking/BookingRepository.cs:26]: is null

File: src/BookingService.Data/Booking/BookingRepository.cs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      {
+        _logger.LogWarning(new ArgumentNullException(nameof(booking)).Message);
+        return;
+      }
+
+      _provider.Bookings.Add(booking);
+      await _provider.SaveAsync();
```
Comment (VLiamin) [inline src/BookingService.Data/Booking/BookingRepository.cs:33]: May be without await?

Comment (SolovyovDmitry) [reply src/BookingService.Data/Booking/BookingRepository.cs:33]: Can't see the reason for this

File: src/BookingService.Business/Commands/Booking/CreateBookingCommand.cs [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+      IHttpContextAccessor httpContextAccessor,
+      IResponseCreator responseCreator)
+    {
+      _accessValidator = accessValidator;
+      _repository = repository;
+      _validator = validator;
+      _mapper = mapper;
+      _httpContextAccessor = httpContextAccessor;
+      _responseCreator = responseCreator;
+    }
```
Comment (MariyaSin) [inline src/BookingService.Business/Commands/Booking/CreateBookingCommand.cs:42]: add empty line

File: src/BookingService.Models.Db/DbBooking.cs [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+  public class DbBookingConfiguration : IEntityTypeConfiguration<DbBooking>
+  {
+    public void Configure(EntityTypeBuilder<DbBooking> builder)
+    {
+      builder.ToTable(DbBooking.TableName);
+
+      builder.HasKey(t => t.Id);
+
+      builder.Property(w => w.WorkspaceId).IsRequired();
```
Comment (MariyaSin) [inline src/BookingService.Models.Db/DbBooking.cs:27]: no need for value type

File: src/BookingService.Models.Db/DbBooking.cs [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+      builder.Property(st => st.StartTime).IsRequired();
```
Comment (MariyaSin) [inline src/BookingService.Models.Db/DbBooking.cs:29]: same

File: src/BookingService.Models.Db/DbBooking.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+      builder.Property(et => et.EndTime).IsRequired();
```
Comment (MariyaSin) [inline src/BookingService.Models.Db/DbBooking.cs:31]: same

File: src/BookingService.Validation/Booking/CreateBookingValidator.cs [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+  public class CreateBookingValidator : AbstractValidator<CreateBookingRequest>, ICreateBookingValidator
+  {
+    public CreateBookingValidator(
+      IBookingRepository bookingRepository,
+      IOfficeService officeService)
+    {
+      RuleFor(r => r)
+        .Cascade(CascadeMode.Stop)
+        .Must(r => r.StartTime != default && r.EndTime != default)
+        .WithMessage("Start and end time must be specified")
```
Comment (MariyaSin) [inline src/BookingService.Validation/Booking/CreateBookingValidator.cs:20]: put a dot at the end of a sentence

File: src/BookingService.Validation/Booking/CreateBookingValidator.cs [hunk 8]
```diff
@@ -0,0 +1,33 @@
...
+        .Must(r => r.StartTime > DateTime.Now && r.EndTime > DateTime.Now)
+        .WithMessage("Start and end time must be after current time")
+        .Must(r => r.StartTime < r.EndTime)
+        .WithMessage("End time must be after start time")
+        .MustAsync(async (r, _) =>
+          await officeService.CheckWorkspaceExistence(r.WorkspaceId, new List<string>()))
```
Comment (MariyaSin) [inline src/BookingService.Validation/Booking/CreateBookingValidator.cs:26]: you can use the method without errors

File: src/BookingService/Startup.cs [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+          ResponseWriter = UIResponseWriter.WriteHealthCheckUIResponse
+        });
+      });
+    }
+
+    #endregion
+
+    #region private methods
+
+    private (string username, string password) GetRabbitMqCredentials()
```
Comment (MariyaSin) [inline src/BookingService/Startup.cs:169]: use kernel method

File: src/BookingService/appsettings.json [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+    ],
+    "WriteTo": [
+      {
+        "Name": "Console"
+      },
+      {
+        "Name": "Seq",
+        "Args": {
+          "serverUrl": "http://localhost:5342",
+          "apiKey": "zxmpuBKg6KnigYgJxW2c"
```
Comment (MariyaSin) [inline src/BookingService/appsettings.json:20]: zrmpuBKg6KnigYgJxW1c

File: src/BookingService.Business/Commands/Booking/CreateBookingCommand.cs [hunk 11]
```diff
@@ -0,0 +1,71 @@
...
+
+    public async Task<OperationResultResponse<Guid?>> ExecuteAsync(CreateBookingRequest request)
+    {
+      if (!await _accessValidator.HasRightsAsync(Rights.AddEditRemoveCompanyData))
```
Comment (MariyaSin) [inline src/BookingService.Business/Commands/Booking/CreateBookingCommand.cs:46]: add toDo "update right"

File: src/BookingService.Data/Booking/BookingRepository.cs [hunk 12]
```diff
@@ -0,0 +1,49 @@
...
+
+      return booking.Id;
+    }
+
+    public async Task<bool> HasOverlapAsync(Guid workspaceId, DateTime startTime, DateTime endTime)
+    {
+      return await _provider.Bookings.AnyAsync(b =>
```
Comment (MariyaSin) [inline src/BookingService.Data/Booking/BookingRepository.cs:40]: remove async\await and return Task

File: src/BookingService/appsettings.Development.json [hunk 13]
```diff
@@ -0,0 +1,22 @@
+{
+  "Serilog": {
+  },
+  "Kestrel": {
+    "EndPoints": {
+      "Http": {
+        "Url": "http://[:]:9835"
```
Comment (MariyaSin) [inline src/BookingService/appsettings.Development.json:7]: 9856

File: src/BookingService/appsettings.Development.json [hunk 14]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  "ConnectionStrings": {
+    "SQLConnectionString": "Server=localhost\\sqlexpress;Database=BookingDB;Trusted_Connection=True;"
+  },
+  "RabbitMQ": {
+    "Host": "localhost",
+    "Username": "guest",
+    "Password": "guest"
+  },
+  "Redis": {
```
Comment (MariyaSin) [inline src/BookingService/appsettings.Development.json:19]: dou you use redis?

File: src/BookingService/appsettings.Production.json [hunk 15]
```diff
@@ -0,0 +1,13 @@
+﻿{
+  "Serilog": {
+  },
+  "ConnectionStrings": {
+    "SQLConnectionString": "Server=dbBooking-sql-server;User=sa;Password=Booking_1234;Database=BookingDB;"
+  },
+  "RabbitMQ": {
+    "Host": "lt_do_rabbitmq"
+  },
+  "Redis": {
```
Comment (MariyaSin) [inline src/BookingService/appsettings.Production.json:10]: same

File: src/BookingService.Broker/Requests/OfficeService.cs [hunk 16]
```diff
@@ -0,0 +1,33 @@
...
+      Response<IOperationResult<bool>> result =
+        await _rcCheckWorkspaceExistence.GetResponse<IOperationResult<bool>>(
+          ICheckWorkspaceIsBookableRequest.CreateObj(workspaceId));
+
+      if (result.Message is null)
+      {
+        return false;
+      }
+
+      return result.Message.Body;
```
Comment (ndoriomedov) [inline src/BookingService.Broker/Requests/OfficeService.cs:30]: @MariyaSin can't we return Task here and check if it is null or not in the command?

**Review by MariyaSin — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**